packages/daemon/src/daemon.js

```diff
@@ -583,6 +618,117 @@ const makeEndoBootstrap = (
     makeMailbox,
   });
 
+  /**
+   * Creates an inspector for the current party's pet store, used to create
+   * inspectors for values therein. Notably, can provide references to otherwise
+   * un-nameable values such as the `MAIN` worker. See `KnownEndoInspectors` for
+   * more details.
+   *
+   * @param {string} petStoreFormulaIdentifier
+   * @returns {Promise<import('./types').EndoInspector>}
+   */
+  const makePetStoreInspector = async petStoreFormulaIdentifier => {
+    const petStore = await provideValueForFormulaIdentifier(
+      petStoreFormulaIdentifier,
+    );
+
+    /**
+     * @param {string} petName - The pet name to inspect.
+     * @returns {Promise<import('./types').KnownEndoInspectors[string]>} An
+     * inspector for the value of the given pet name.
+     */
+    const lookup = async petName => {
```

Comment:
We have types for the individual known inspectors (see `types.d.ts`), but due to the indirection between the `petName` argument and the `formulaType`, it is difficult (impossible?) to map the `petName` to the "correct" return type.

Reply:
Agreed. We can’t be more specific than a union of all known inspector types. Lookup in general has this limitation. For some Endo APIs, we could overload a formula type argument and provide guarantees of specific types, to make some casts unnecessary.